libbpf-cargo/src/btf/btf.rs

```diff
@@ -380,27 +408,88 @@ impl<'a> Btf<'a> {
                             )?;
 
                             if padding != 0 {
-                                writeln!(
-                                    def,
+                                agg_content.push(format!(
                                     r#"    __pad_{offset}: [u8; {padding}],"#,
                                     offset = offset,
                                     padding = padding,
-                                )?;
+                                ));
+
+                                impl_default.push(format!(
+                                    r#"            __pad_{offset}: [u8::default(); {padding}]"#,
+                                    offset = offset,
+                                    padding = padding,
+                                ));
+                            }
+
+                            match self.type_by_id(field_ty_id)? {
+                                BtfType::Array(ft) => {
+                                    if ft.nelems > 32 {
+                                        gen_impl_default = true
+                                    }
+                                }
+                                _ => {}
                             }
+
+                            match self.type_default(field_ty_id) {
+                                Ok(def) => {
+                                    impl_default.push(format!(
+                                        r#"            {field_name}: {field_ty_str}"#,
+                                        field_name = member.name,
+                                        field_ty_str = def
+                                    ));
+                                }
+                                Err(_) => {
+                                    if gen_impl_default {
+                                        bail!("Could not construct a necessary Default Impl");
```

Comment:
More helpful error message
```suggestion
                                Err(e) => {
                                    if gen_impl_default {
                                        bail!("Could not construct a necessary Default Impl: {}", e);
```